src/python/foglamp/device/ingest.py

```diff
@@ -118,14 +299,40 @@ def increment_discarded_readings(cls):
                 cls._discarded_readings = 0
             # TODO catch real exception
             except Exception:
-                _LOGGER.exception("An error occurred while writing readings statistics")
+                _LOGGER.exception('An error occurred while writing readings statistics')
+
+        _LOGGER.info('Device statistics writer stopped')
+
+    @classmethod
+    def is_available(cls) -> bool:
```

Comment:
instead of doing this, we should keep a queue keep consumed if overflowing then log and halt ?!

Reply:
At some point you have to stop queueing.

This currently happens only if the database connection goes down. I can't make it happen by bombarding the server with requests - something is holding down the number of requests that are sent to the handler. There appears to be a queue (probably the event loop) of calls to the aiocoap handler. If the handler doesn't process each request in a consistent amount of time, this queue will grow without bound until the process is out of memory.

How to respond to database connection errors (or perhaps the database is out of disk space) requires more thought.